packages/cxl-ui/src/components/cxl-playbook-accordion.js

```
@@ -0,0 +1,163 @@
+
+      items.forEach((value, key) => {
+        stateItems[key] = items[key].opened;
+      });
+
+      localStorage.setItem(storageId, JSON.stringify(stateItems));
+
+      this._updateCSSAndPanelStateToCheckboxesStates();
+
+      this._dispatchCustomEvent(stateItems);
+
+      this._saveCheckboxesState();
+    }
+  }
+
+  _saveCheckboxesState() {
+    const stateCheckboxes = [];
+    const checkboxes = this.querySelectorAll('vaadin-checkbox');
+
+    checkboxes.forEach((value, key) => {
+      const checkbox = checkboxes[key];
+
+      stateCheckboxes[key] =
+        checkbox.hasAttribute('aria-checked') && checkbox.getAttribute('aria-checked') === 'true';
+    });
+
+    const checkboxesStorageId = `${this.getAttribute('id')}_checkboxes`;
+    localStorage.setItem(checkboxesStorageId, JSON.stringify(stateCheckboxes));
+  }
+
+  // Restore accordion panel state.
+  _updateItems(items, opened) {
+    super._updateItems(items, opened);
+  }
+
+  _updateCheckboxesStates() {
+    const checkboxesStorageId = `${this.getAttribute('id')}_checkboxes`;
+    const stateCheckboxes = JSON.parse(localStorage.getItem(checkboxesStorageId));
+
+    if (stateCheckboxes === null) {
+      return;
+    }
+
+    const checkboxes = this.querySelectorAll('vaadin-checkbox');
+
+    checkboxes.forEach((item, key) => {
+      const checkbox = checkboxes[key];
+      const isChecked = !!stateCheckboxes[key]; // autocast from null or undefined to boolean
+      checkbox.setAttribute('aria-checked', isChecked ? 'true' : 'false');
+      checkbox.checked = isChecked;
+    });
+
+    this._updateCSSAndPanelStateToCheckboxesStates();
+  }
+
+  _updateCSSAndPanelStateToCheckboxesStates() {
+    const checkboxes = this.querySelectorAll('vaadin-checkbox');
+    const panels = this.querySelectorAll('vaadin-accordion-panel');
+
+    checkboxes.forEach((checkbox, index) => {
+      if (!panels[index])
+        throw new Error(`vaadin-accordion-panel with index ${index} doesn't exist`);
```

What's the currently known scenarios for such index mismatches?

---

Maybe it's more clear to ask: when does this Error potentially happen?

---

> Maybe it's more clear to ask: when does this Error potentially happen?

Previously selectors for **this.panelSummarySlotCheckboxes**  and **this.accordionPanels** were independent (this may well be changed in the future). So there is an informational exception in case any checkbox does not have a corresponding panel. In case of any possible future debugging, I prefer to throw transparent exception info, instead of leaving possible `Undefined error`.